hs_core/hydroshare/resource.py

```diff
@@ -478,29 +483,39 @@ def update_resource(
         resource.owners.add(owner)
 
     if edit_users:
+        # TODO: remove edit_users
         resource.edit_users.clear()
         for user in edit_users:
             user = utils.user_from_id(user)
+            # TODO remove edit_users
             resource.edit_users.add(user)
+            # TODO: remove view_users
             resource.view_users.add(user)
 
     if view_users:
+        # TODO: remove view_users
         resource.view_users.clear()
         for user in view_users:
             user = utils.user_from_id(user)
+            # TODO: remove view_users
             resource.view_users.add(user)
 
     if edit_groups:
+        # TODO: remove edit_groups
         resource.edit_groups.clear()
         for group in edit_groups:
             group = utils.group_from_id(group)
             resource.edit_groups.add(group)
+            # TODO: remove view_groups
             resource.view_groups.add(group)
 
+    # FIXME: why does checking view_groups lead to mutating edit_groups & view_groups?
     if view_groups:
+        # TODO: remove edit_groups
         resource.edit_groups.clear()
         for group in view_groups:
             group = utils.group_from_id(group)
+            # TODO: remove view_groups
```

Comment:
What is the state of this work with all these TODO left undone? Are there features broken or incomplete until this is done?